components/kserve/servicemesh_setup.go

```diff
@@ -13,42 +13,47 @@ import (
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
 )
 
-func (k *Kserve) configureServiceMesh(c client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+func (k *Kserve) configureServiceMesh(cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
 	if dscispec.ServiceMesh.ManagementState == operatorv1.Managed && k.GetManagementState() == operatorv1.Managed {
-		serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures(c))
+		serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec.ApplicationsNamespace, k.defineServiceMeshFeatures(cli, dscispec))
 		return serviceMeshInitializer.Apply()
 	}
 	if dscispec.ServiceMesh.ManagementState == operatorv1.Unmanaged && k.GetManagementState() == operatorv1.Managed {
 		return nil
 	}
 
-	return k.removeServiceMeshConfigurations(c, dscispec)
+	return k.removeServiceMeshConfigurations(cli, dscispec)
 }
 
 func (k *Kserve) removeServiceMeshConfigurations(cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
-	serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures(cli))
+	serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec.ApplicationsNamespace, k.defineServiceMeshFeatures(cli, dscispec))
 	return serviceMeshInitializer.Delete()
 }
 
-func (k *Kserve) defineServiceMeshFeatures(cli client.Client) feature.FeaturesProvider {
-	return func(handler *feature.FeaturesHandler) error {
+func (k *Kserve) defineServiceMeshFeatures(cli client.Client, dscispec *dsciv1.DSCInitializationSpec) feature.FeaturesProvider {
+	return func(registry feature.FeaturesRegistry) error {
 		authorinoInstalled, err := cluster.SubscriptionExists(cli, "authorino-operator")
 		if err != nil {
 			return fmt.Errorf("failed to list subscriptions %w", err)
 		}
 
 		if authorinoInstalled {
-			kserveExtAuthzErr := feature.CreateFeature("kserve-external-authz").
-				For(handler).
+			kserveExtAuthzErr := registry.Add(feature.Define("kserve-external-authz").
 				ManifestsLocation(Resources.Location).
 				Manifests(
 					path.Join(Resources.ServiceMeshDir, "activator-envoyfilter.tmpl.yaml"),
 					path.Join(Resources.ServiceMeshDir, "envoy-oauth-temp-fix.tmpl.yaml"),
 					path.Join(Resources.ServiceMeshDir, "kserve-predictor-authorizationpolicy.tmpl.yaml"),
 					path.Join(Resources.ServiceMeshDir, "z-migrations"),
 				).
-				WithData(servicemesh.ClusterDetails).
-				Load()
+				WithData(
+					feature.Entry("Domain", cluster.GetDomain),
```

Comment:
Does GetDomain possible belong together with some other "common" data? Maybe we could have collected them under something like system/cluster.FeatureData() ?

Reply:
Yeah, that makes sense. I will work on moving it somewhere, as it used in several places.

Reply:
It is used only in one place right now, keeping it as simple `Entry`